File: VioletMarketplace/src/main/java/com/skilldistillery/marketplace/services/TokenService.java
package com.skilldistillery.marketplace.services;

import java.util.Set;

import com.skilldistillery.marketplace.entities.Token;
import com.skilldistillery.marketplace.dto.TokenUpdateRequest;

public interface TokenService {
	Set<Token> index();

	Set<Token> indexByUsername(String username);

	Token findById(int tid);

	Token create(String username, Token token);

	Token purchase(String buyerName, int tid);

	Token update(String ownerName, TokenUpdateRequest request);

	boolean destroy(String username, int tid);

    boolean tokenExists(int tid);

	boolean userOwnsToken(String username, int tid);

}

File: VioletMarketplace/src/main/java/com/skilldistillery/marketplace/services/TokenService.kt
package com.skilldistillery.marketplace.services

import com.skilldistillery.marketplace.dtos.TokenUpdateRequest
import com.skilldistillery.marketplace.entities.Token

interface TokenService {
    fun index(): Set<Token>
    fun indexByUsername(username: String): Set<Token>
    fun findById(tid: Int): Token
    fun create(username: String, token: Token): Token
    fun purchase(buyerName: String, tid: Int): Token
    fun update(ownerName: String, request: TokenUpdateRequest): Token
    fun destroy(username: String, tid: Int): Boolean
    fun tokenExists(tid: Int): Boolean
    fun userOwnsToken(username: String, tid: Int): Boolean
}
